### DISCUSSION TITLE
How to get MQTT client ID from a delivery consumed over AMQP 0-9-1 or AMQP 1.0?

### DISCUSSION THREAD

**Discussion by mthgh0818** [rabbitmq/rabbitmq-server#9062] [Other] [answered]
### Describe the bug

I use amqp listen the mqtt messgae,but the message payload not contins client id, how to get the client id by amqp?

### Reproduction steps

1. nothing
2.
3.
...

### Expected behavior

I use amqp listen the mqtt messgae,but the message payload not contins client id, how to get the client id by amqp?

**Comment by michaelklishin** [comment, marked as answer]
There is no client ID equivalent in either AMQP 0-9-1 or AMQP 1.0, so the property is not propagated. It could be set to a header value but so far, there hasn't been a lot of demand for that.

**Reply by ansd** [reply]
Doesn't MQTT client ID map to AMQP 1.0 user ID and AMQP 0.9.1 user ID?
Although in MQTT there's both username and client ID which are two different things.

We'll get better and more defined protocol conversion semantics with https://github.com/rabbitmq/rabbitmq-server/pull/5077.
That said, on that branch an MQTT client ID isn't translated into AMQP user ID either.

The client ID is not part of the MQTT application message. It belongs to the whole MQTT session. Inserting MQTT session related information into every message from other protocols upon translation / consuming the message doesn't make sense to me.

If you are in control of the applications, you could insert the MQTT client ID into the User Property of the MQTT PUBLISH packet which will then be translated into AMQP 1.0 Application Properties.

**Reply by mthgh0818** [reply]
Thanks Reply，I want know how to set the clientId in the header
